lib/browser/api/net.ts

```diff
@@ -526,6 +529,122 @@ export function request (options: ClientRequestConstructorOptions | string, call
   return new ClientRequest(options, callback);
 }
 
+function createDeferredPromise<T, E extends Error = Error> (): { promise: Promise<T>; resolve: (x: T) => void; reject: (e: E) => void; } {
+  let res: (x: T) => void;
+  let rej: (e: E) => void;
+  const promise = new Promise<T>((resolve, reject) => {
+    res = resolve;
+    rej = reject;
+  });
+
+  return { promise, resolve: res!, reject: rej! };
+}
+
+export function fetch (input: RequestInfo, init?: RequestInit): Promise<Response> {
```

Comment:
Are the `RequestInfo` and `RequestInit` types coming from Node.js ?

Reply:
yep.

Reply:
actually, it looks like these types are coming from `lib.dom.d.ts` in Typescript. @types/node doesn't export them, though undici does define them: https://github.com/nodejs/undici/blob/2b260c997ad4efe4ed2064b264b4b546a59e7a67/types/fetch.d.ts#L103

This turns out to cause some minor issues (e.g. TypeScript getting confused between the Web's `ReadableStream` and Node's Web-compatible `ReadableStream`) but on the whole the types should be pretty similar, so not too big a deal I think, at least until Node sorts out what it's doing w.r.t. fetch &c.